### ISSUE TITLE
error 'someVal' is defined but never used no-unused-vars[rulename]

### ISSUE THREAD

**Issue by UglyHobbitFeet** [typescript-eslint/typescript-eslint#3972]
- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["off"]
  }
}
```

```TS
// your repro code case
I am getting the error mentioned in the title when I have the following:

type MyType = (someVal: string) => string | true;
```

Can't. It's long and I can't cut/paste so would have to hand-type it.

**Expected Result**

That types would be ignored for this case

**Actual Result**

They are not

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.33.0` |
| `@typescript-eslint/parser`        | `4.33.0` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `6.7.2` |
| `node`                             | `14.17.6` |

**Comment by bradzacher** [comment]
```json
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["off"]
  }
}
```

You list your config as having our version of the rule turned off, and you list the error message as

```
'someVal' is defined but never used no-unused-vars
                                    ^^^^^^^^^^^^^^
```

The key bit is the last bit that I've highlighted.
It's not our lint rule that you're using - it's the base ESLint one which does not support TypeScript.

Which is why in your last issue I linked you directly to the FAQ article which solves your issue.
https://github.com/typescript-eslint/typescript-eslint/issues/3971#issuecomment-934602732

You need to use our version of the rule.

This is also listed in our docs for the rule.
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md#how-to-use
